document/RuleMetadataProfile.mdk

```diff
@@ -1 +1,165 @@
+When present, the `options` objects shall describe the set of configurable options supported by the rule.
+
+The value within each string-value pair shall be a string, which may be the empty string.
+The value shall not be a JSON array or sub-object.
+
+### `formatSpecifiers` property { #ruleDescriptor-formatSpecifiers }
+
+A `ruleDescriptor` object may contain a property named `formatSpecifiers`, whose value is a JSON object
+consisting of an arbitrary set of string/value pairs.
+When present, the `formatSpecifiers` object shall describe a set of message formatting strings
+that have or may be returned by the tool.
+The string name of each format string shall be an identifier that uniquely identifies each member.
+The value within each string-value pair shall be a string that can be passed as a control parameter to
+string formatting API (_e.g._, `printf`) in order to construct a formatted message in combination with
+an arbitrary number of additional data parameters.
+
+The `formatSpecifiers` property contents may be limited to values that are associated with issues
+written to the issue log.
+In this use, the relevant issue objects must provide a `formattedMessage` (&sect;[#formattedMessage]) object
+that refers to an associated messageFormatString. In another use, the `formatSpecifiers` property 
+contents may be used to document all possible message format strings that can be returned by a rule.
+
+~ IsoExample
+
+````
+{
+    "objectCreation" : "{0} creates a new instance of {1} which is never used.
+	                    Pass the instance as an argument to another method, assign the instance to a variable,
+						or remove the object creation if it is unnecessary.",  
+    "stringReturnValue" : "{0} calls {1} but does not use the new string instance that the method returns.
+	                       Pass the instance as an argument to another method, assign the instance to a variable,
+						  or remove the call if it is unnecessary."        
+}
+````
+~
+
+### `properties` property { #ruleDescriptor-properties }
+
+A `ruleDescriptor` object may contain a property named `properties`, whose value is a JSON object
+consisting of an arbitrary set of string/value pairs.
+This allows tools to include information about the rule that is not explicitly specified in the SARIF format.
+
+The value within each string-value pair shall be a string, which may be the empty string.
+The value shall not be a JSON array or sub-object.
+
+## `issue` object { #rule-metadata-profile-issue }
+
+### General { #rule-metadata-profile-issue-general }
+
+The Rule Metadata Profile defines one additional property in the `issue` object (&sect;[#issue]).
+
+### `formattedMessage` property { #issue-formattedMessage }
+
+An `issue` object (&sect;[#issue]) may contain a `formattedMessage` property whose value is
+a `formattedMessage` object (&sect;[#formattedMessage]) that can be used to construct
+a fully formatted message that describes the issue.
+
```

Comment:
yes, tools would need to provide support for this. at least one tool, fxcop, emits its output in this way. i'm not aware of others that do. fwiw, this scenario also supports producing output that can be evaluated as a post-processing step. i.e., by separating the dynamic data from the static formatting string, you can better enable analysis of the dynamic data itself. e.g., you could emit the compiler version used to produce a binary as standalone data, then retroactively apply policy to the output file which ensures that the version meets or exceeds some min-bar (for security or other reasons)